### ISSUE TITLE
Server Side Socket Programming

### ISSUE THREAD

**Issue by yogesh-1303** [cu-ecen-aeld/final-project-yogesh-1303-app#2]
To program the server side of the socket programming

DoD: The server socket is working and transferring some test data (random characters) to the client, on the buildroot system.

Initially Blocked by:
[Server Buildroot Setup](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-buildroot/issues/1)

More Blockers: Connecting with client, testing the program on the host system, cross compiling and running.

**Comment by yogesh-1303** [comment]
Successful cross-compilation of a simple c program (test_hello.c) with successful run on Rpi system with buildroot has been accomplished. 
Following are the images:
[Cross-compilation from host system](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/blob/main/cross_compilation.PNG)
[successful run on buildroot system](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/blob/main/buildroot_test_run_successful.PNG)

**Comment by yogesh-1303** [comment]
The server is sending a string to client successfully to the client on buildroot system. The test string is shown in the proof images present on [this link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/tree/main/client-server%20images).

During the process of completing this issue some blockers were faced. On of those blockers was finding out the correct toochain forthe raspberry pi buildroot system. I had to try different toolchans and deal with an error. The link to the blocker error is given below:
[Toolchain Blocker Link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/blob/main/client-server%20images/error.PNG)

Other than this error, there were several programming errors that I had to debug.
As far as procedure is concerned, I started with finding out the toolchain for raspberry pi buildroot. Then I tried implement the server program on the local host using the socket.sh to test whether it is connecting and sending some data to the client. After that I followed with the server side programming.

I the final coming together of the component of the project, I will implement Daemon feature and include a shell script to operate the server executable. Also I will link this repository using shell script with the buildroot repository.

As per the DoD, the issue is completed successfully. Below is the link for the proof images for the completed issue:
[Issue completed](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/tree/main/client-server%20images)

Below is the commit link for the issue:
[commit link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/commit/a65a32620cf608753346bc0ed94c7e35264ce1f0)

**Comment by yogesh-1303** [comment]
The server is sending a string to client successfully to the client on buildroot system. The test string is shown in the proof images present on [this link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/tree/main/client-server%20images).

During the process of completing this issue some blockers were faced. On of those blockers was finding out the correct toochain forthe raspberry pi buildroot system. I had to try different toolchans and deal with an error. The link to the blocker error is given below:
[Toolchain Blocker Link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/blob/main/client-server%20images/error.PNG)

Other than this error, there were several programming errors that I had to debug.
As far as procedure is concerned, I started with finding out the toolchain for raspberry pi buildroot. Then I tried implement the server program on the local host using the socket.sh to test whether it is connecting and sending some data to the client. After that I followed with the server side programming.

I the final coming together of the component of the project, I will implement Daemon feature and include a shell script to operate the server executable. Also I will link this repository using shell script with the buildroot repository.

As per the DoD, the issue is completed successfully. Below is the link for the proof images for the completed issue:
[Issue completed](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/tree/main/client-server%20images)

Below is the commit link for the issue:
[commit link](https://github.com/cu-ecen-aeld/final-project-yogesh-1303-app/commit/a65a32620cf608753346bc0ed94c7e35264ce1f0)
